pkg/cnab/config-adapter/adapter.go

```diff
@@ -309,6 +305,70 @@ func (c *ManifestConverter) generateDependencies() *extensions.Dependencies {
 	return deps
 }
 
+func (c *ManifestConverter) generateParameterSources(b *bundle.Bundle) extensions.ParameterSources {
+	ps := extensions.ParameterSources{}
+
+	// Parameter sources come from two places, indirectly from our template wiring
+	// and directly when they use `source` on a parameter
+
+	for _, p := range c.Manifest.Parameters {
+		if p.Source.Output == "" {
+			continue
+		}
+
+		pso := c.generateParameterSource(p.Source.Output)
+		ps[p.Name] = pso
+	}
+
+	for _, outputDef := range c.Manifest.GetTemplatedOutputs() {
+		wiringName, p, def := c.generateOutputWiringParameter(*b, outputDef.Name)
+		if b.Parameters == nil {
+			b.Parameters = make(map[string]bundle.Parameter, 1)
+		}
+		b.Parameters[wiringName] = p
+		b.Definitions[wiringName] = &def
+
+		pso := c.generateParameterSource(outputDef.Name)
+		ps[wiringName] = pso
+	}
+
+	return ps
+}
+
+func (c *ManifestConverter) generateOutputWiringParameter(b bundle.Bundle, outputName string) (string, bundle.Parameter, definition.Schema) {
+	wiringName := manifest.GetParameterSourceName(outputName)
+
+	wiringParam := bundle.Parameter{
+		Definition:  wiringName,
+		Description: fmt.Sprintf("Wires up the %s output for use as a parameter. Porter internal parameter that should not be set manually.", outputName),
+		Required:    false,
+		Destination: &bundle.Location{
+			EnvironmentVariable: manifest.GetParameterSourceEnvVar(outputName),
+		},
+	}
+
+	// Copy the output definition for use with the wiring parameter
+	// and identify the definition as a porter internal structure
+	outputDefName := b.Outputs[outputName].Definition
+	outputDef := b.Definitions[outputDefName]
+	var wiringDef definition.Schema
+	wiringDef = *outputDef
+	wiringDef.ID = "https://porter.sh/schema/bundle.json#porter-parameter-source-definition"
```

Comment:
Same as above, these are meant just for porter to say "this is mine", not for schema / resolution. Does ID have to be a resolveable URI? My intent was to create a string that was unambiguous on who generate the field and what it is used for so that we are sure it is our wiring parameter source. I wasn't sure it was strictly necessary but it seemed like a useful thing to put in now rather than regret later.

If I should be using a different field in the json schema, let me know. There isn't a schema to back it, it conforms to bundle.json. It just says who generated it, you know?